### ISSUE TITLE
Transparency on Linux

### ISSUE THREAD

**Issue by sant123** [dracula/zed#16]
Hi, not sure if this is expected behavior but in Visual Studio Code there is no transparency in the files tab.

Zed
![image](https://github.com/dracula/zed/assets/7959437/f30d8cde-7195-4f64-9554-1ca0fe6f6938)

Code
![image](https://github.com/dracula/zed/assets/7959437/63147cbe-a66a-41bf-bada-d26de38307cb)

Thanks

**Comment by sant123** [comment]
I'm using Fedora 40

**Comment by sant123** [comment]
Tested on Mac and works good 
<img width="1728" alt="image" src="https://github.com/dracula/zed/assets/7959437/865296b3-ad90-48d9-8c74-b874c6ea04ae">

**Comment by lsunsi** [comment]
Same here! Just installed it on Linux and transparency was definitely unexpected.

**Comment by black-puppydog** [comment]
Running debian unstable with latest updates from today, on X11 and an AMD Vega mobile GPU.
Transparency in the sidebar makes it pretty unusable. For the time being, this is gonna be my ownly non dracula themed editor 😢

**Comment by smallSwed** [comment]
I have the same transparency issue on Ubuntu 24.04 with wayland.

**Comment by Otterpocket** [comment]
```json
  "experimental.theme_overrides": {
    "background.appearance": "opaque"
  },
```

Try putting this in the settings file.

**Comment by Otterpocket** [comment]
Also, sorry for taking so long to respond.... GitHub notifications..

Ok so I can explain what is happening but not why.

A feature got released in zed to accommodate for transparent/blurred UI, by adding the setting above with one of the values:  `opaque`, `blurred` or `transparent`.
For these features to do anything, parts of the theme (such as the project panel) need a 8 digit hex notation to define transparency.
This should only come into effect when blurred or transparent are set.
This has been part of the theme for quite a while now and has always worked well on Mac.

So my conclusion is that this is more of a bug with the linux release than the theme itself.

I'll look for any bug reports on the zed repo and link them here.

**Comment by lsunsi** [comment]
@Otterpocket I just tried it, but it looks the same. Maybe I'm not placing it right

**Comment by lisotton** [comment]
I'm also having this issue. I'm on Pop_OS 22.04. 
I tried the solution provided by @Otterpocket, but did not work.

**Comment by dnikoloski** [comment]
I am affected by this on Bluefin, fixed with this:

```
  "experimental.theme_overrides": {
    "background": "#282a36ff",
    "title_bar.background": "#141119"
  }
```

**Comment by smallSwed** [comment]
> I am affected by this on Bluefin, fixed with this:
> 
> ```
>   "experimental.theme_overrides": {
>     "background": "#282a36ff",
>     "title_bar.background": "#141119"
>   }
> ```

this is a good workaround, not 100% because when the Zed is out of focus the title bar and the status bar still transparent, but those can fixed also this way:

```json
{
  "experimental.theme_overrides": {
    "background": "#282a36ff",
    "title_bar.background": "#141119ff",
    "title_bar.inactive_background": "#1c1d26ff",
    "status_bar.background": "#141119ff"
  }
}
```

**Comment by alvinseptiano** [comment]
I'm using currently using Ubuntu my workaround for this by pressing `Windows / Super` + `M` key.

**Comment by Otterpocket** [comment]
<img width="691" alt="image" src="https://github.com/user-attachments/assets/bb240dc4-5d74-4188-9b70-35befa2ede71">
This was in the latest preview release, does it resolve this issue?

**Comment by smallSwed** [comment]
unfortunately no, the side bar and the title bar still transparent: 
![Screenshot from 2024-08-20 10-36-17](https://github.com/user-attachments/assets/ec872cb9-0986-4877-81f8-8898315c9020)

**Comment by Otterpocket** [comment]
> unfortunately no, the side bar and the title bar still transparent: ![Screenshot from 2024-08-20 10-36-17](https://private-user-images.githubusercontent.com/29921902/359382111-ec872cb9-0986-4877-81f8-8898315c9020.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxNDIxMjUsIm5iZiI6MTcyNDE0MTgyNSwicGF0aCI6Ii8yOTkyMTkwMi8zNTkzODIxMTEtZWM4NzJjYjktMDk4Ni00ODc3LTgxZjgtODg5ODMxNWM5MDIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIwVDA4MTcwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzZDAyODE5YWUzMzY4NWJlNjE5MTg5ODU1NmYyMWUwYjhkNGYxZWIzNDlmZWVhZTcwN2E4ZTNmOTc0NTc3ZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.F5U-n3oW9VxA7JCjN_XqeDrQOlZDZExeuHFvhaJS7Nc)

and:

```json
  "experimental.theme_overrides": {
    "background.appearance": "opaque"
  },
```

still doesn't work?

**Comment by smallSwed** [comment]
nope, it does not work. It seems like that setting is ignored, nothing changes when setting to any of the valid values ("opaque", "transparent" or "blurred")

**Comment by JonasAlv** [comment]
> > unfortunately no, the side bar and the title bar still transparent: ![Screenshot from 2024-08-20 10-36-17](https://private-user-images.githubusercontent.com/29921902/359382111-ec872cb9-0986-4877-81f8-8898315c9020.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxNDIxMjUsIm5iZiI6MTcyNDE0MTgyNSwicGF0aCI6Ii8yOTkyMTkwMi8zNTkzODIxMTEtZWM4NzJjYjktMDk4Ni00ODc3LTgxZjgtODg5ODMxNWM5MDIwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIwVDA4MTcwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzZDAyODE5YWUzMzY4NWJlNjE5MTg5ODU1NmYyMWUwYjhkNGYxZWIzNDlmZWVhZTcwN2E4ZTNmOTc0NTc3ZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.F5U-n3oW9VxA7JCjN_XqeDrQOlZDZExeuHFvhaJS7Nc)
> 
> and:
> 
> ```json
>   "experimental.theme_overrides": {
>     "background.appearance": "opaque"
>   },
> ```
> 
> still doesn't work?

mine is working i guess 
![image](https://github.com/user-attachments/assets/0e26d11a-630e-4832-b75e-3e2d3311b11d)

**Comment by lisotton** [comment]
The suggestion from @JonasAlv worked for me.

**Comment by Otterpocket** [comment]
Yes, Overriding the colours will certainly work as a temporary fix. I'll leave this ticket open until the actual source of the problem is fixed

**Comment by smallSwed** [comment]
I checked out the dracula.json and found this: https://github.com/dracula/zed/blob/c60784cf4d98d2e3e657218916e92196199d5b8a/themes/dracula.json#L40

Why is the title_bar.background and others have transparency?
        "status_bar.background": "#141119**bb**",
        "title_bar.background": "#141119**bb**",

I think that is the source of the problem.

**Comment by Otterpocket** [comment]
> I checked out the dracula.json and found this:
> 
> https://github.com/dracula/zed/blob/c60784cf4d98d2e3e657218916e92196199d5b8a/themes/dracula.json#L40
> 
> Why is the title_bar.background and others have transparency? "status_bar.background": "#141119**bb**", "title_bar.background": "#141119**bb**",
> 
> I think that is the source of the problem.

That's meant to be there for the blurred and transparency feature to work at all. 
The problem is the feature doesn't work properly on linux.
The linux version was released after the transparency feature was introduced.

![image](https://github.com/user-attachments/assets/c37c4674-58c8-473a-b734-dbbeab76f8c7)
This is what it should look like. (Mac)

**Comment by tripplehelix** [comment]
> This is what it should look like. (Mac)

Could you possibly release a version without the transparencies for those of us who don't use them?

**Comment by CodeWithAlamin** [comment]
I'm also facing this issue on TUXEDO OS Plasma 6.1. But [zero-stacks](https://github.com/zero-stacks)'s workaround worked for me.

**Comment by obarahona10** [comment]
Same issue here, using Fedora 40 with Wayland.

I'll apply the workaround from [@zero-stacks](https://github.com/dracula/zed/issues/16#issuecomment-2311198035) while this gets fixed!

**Comment by Cutieguwu** [comment]
Ok, so I just spent a day digging through Zed's code running in circles, to not much avail. I'm wondering if this issue may be in Blade, but my understanding is limited enough as it is just looking for differences in the platform code.

So far as I can tell, Blur support on X11 is questionable, if even existent.

Dracula's theme currently doesn't define `"background.appearance": "blurred"` thus the default that's put in place when loaded is `WindowBackgroundAppearance::Opaque`. Looking at the [Indigo theme source](https://github.com/p3rception/Indigo-zed/blob/main/themes/indigo.json), specifically the "Indigo | Clear" theme, `"background.appearance": "blurred"` is defined and the blurring works on Wayland.

For whatever reason, testing on Wayland using an override, Opaque will display exactly like Transparent.

My best guess is that on Wayland, an alpha value is always being passed somewhere, and thus the logic for opaque is the same as transparent. That's hard to test of course, as trying the other way around, overriding an opaque theme with transparency, is complicated by just about every value in an opaque theme having an alpha of 'ff'. I have no clue about that though; merely speculation.

However, given the knowledge that the fill-in default is Opaque, we can assume that for some reason, MacOS treats the theme as blurred. Maybe because of a non-solid alpha, it changes to rendering as blurred; again speculation. I don't have a Mac device to test out the logic using different themes.

Anyway, I quickly made a PR to specify to Zed that the theme should be treated as blurred.

**Comment by Cutieguwu** [comment]
@Otterpocket Yes, specifying to Zed to blur the theme didn't make it display exactly like the MacOS example, but it was significantly improved.

It appears that the Opaque mode under linux doesn't override the alpha values in the theme, but again I can't test how MacOS behaves in this scenario.

I might re-propose the idea of having an Opaque theme using solid alpha and a Blurred theme specifying `"background.appearance": "blurred"` as a solution to work around this issue, if anyone wants to use an opaque theme instead of blurred.

Here's Zed with blurring specified using my patch:
Note: The menu is bar blurred the exact same as the MacOS example when in focus.

![Image](https://github.com/user-attachments/assets/4fdf9a3c-7714-4f3c-8811-394e05d935ab)


Graphics Platform: Wayland
DE: KDE Plasma 6.3.5

Zed: Flatpak 0.189.5

**Comment by Otterpocket** [comment]
Ok, so I have just added an opaque version of the theme. "Dracula Solid".
https://github.com/zed-industries/extensions/pull/2894

It's worth noting the entire original theme is made from lots of transparency/shading, this new one is fully opaque in every place other than the button backgrounds and scrollbar.

So let me know if there are any issues with those elements.

It will be available once zed merges it.

<img width="1237" alt="Image" src="https://github.com/user-attachments/assets/08d1bca4-2eee-4e79-a71b-432d6189ac21" />

I'll close this issue once the theme is available.

**Comment by Otterpocket** [comment]
Solid variant now available in Zed extensions
